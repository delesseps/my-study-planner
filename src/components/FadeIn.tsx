//Display fade animation after loading
import * as React from "react";
import styled, { keyframes } from "styled-components";

const FadeIn = ({ children }: { children: React.ReactElement }) => {
  return <Wrapper>{children}</Wrapper>;
};

const fadeIn = keyframes`
   from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

const Wrapper = styled.div`
  animation: ${fadeIn} 0.5s;
`;

export default FadeIn;
