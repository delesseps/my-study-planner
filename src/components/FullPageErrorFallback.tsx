import React from "react";
import { AxiosError } from "axios";
import { Result, Button } from "antd";
import { Link } from "react-router-dom";
import styled from "styled-components";

interface Props {
  error?: Error;
  requestError?: AxiosError;
}

const errors: Record<string, string> = {
  401: "Sorry, you are not authorized to do this operation.",
  500: "Sorry, the server is down. The sadness...",
};

const FullPageErrorFallback = ({ error, requestError }: Props) => {
  if (error) {
    return (
      <Styles.Result
        status={"error"}
        title={"Something Went Wrong"}
        subTitle={"A fatal error has ocurred. Please refresh."}
        extra={
          <Link to="/">
            <Button type="primary">Back Home</Button>
          </Link>
        }
      />
    );
  }

  const errorCode = requestError?.response?.status as any;

  return (
    <Styles.Result
      status={errorCode === 500 ? errorCode : "error"}
      title={errorCode ? errorCode : "Error"}
      subTitle={
        errors[errorCode]
          ? errors[errorCode]
          : "A fatal error has ocurred. Please refresh."
      }
      extra={
        <Link to="/">
          <Button type="primary">Back Home</Button>
        </Link>
      }
    />
  );
};

const Styles = {
  Result: styled(Result)`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    min-height: 100vh;
  `,
};

export default FullPageErrorFallback;
