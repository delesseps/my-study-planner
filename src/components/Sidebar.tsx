import React from "react";
import styled, { keyframes } from "styled-components";
import { CalendarOutlined, HomeOutlined } from "@ant-design/icons";
import { NavLink } from "react-router-dom";
import { breakpoints } from "theme";

const Sidebar: React.FC = props => {
  return (
    <Wrapper>
      <Item activeClassName="active" to="/dashboard" exact>
        <HomeIcon />
        <Text>Home</Text>
      </Item>
      <Item activeClassName="active" to="/dashboard/schedule">
        <CalendarIcon />
        <Text>Schedule</Text>
      </Item>
    </Wrapper>
  );
};

const fadeIn = keyframes`
   from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

const fadeInUp = keyframes`
    0% {
      transform: translateY(10%);      
    }

    100% {
      transform: translateY(0);     
    }
`;

const fadeInUpText = keyframes`
    0% {
      transform: translateY(50%);
      opacity: 0;
    }

    100% {
      transform: translateY(0);
      opacity: 1;
    }
`;

const Wrapper = styled.nav`
  background-color: ${props => props.theme.sidebarBackgroundColor};
  height: 100%;
  width: 8%;
  border-right: 2px solid ${props => props.theme.fontColors.textRgba(0.1)};

  position: fixed;

  @media only screen and (max-width: ${breakpoints.bpMedium}) {
    bottom: 0;
    top: 0;
    height: auto;
    width: 100%;

    position: sticky;

    border-right: none;
    border-top: 1px solid ${props => props.theme.fontColors.textRgba(0.1)};
    padding: 1rem 0;

    flex-direction: row;
  }

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  & > a:not(:last-child) {
    margin-bottom: 3.5rem;

    @media only screen and (max-width: ${breakpoints.bpMedium}) {
      margin-bottom: 0;
    }
  }

  .active {
    & p {
      display: initial;
      animation: ${fadeInUpText} ease-out 0.3s;
      margin-bottom: -1.3rem;

      @media only screen and (max-width: ${breakpoints.bpMedium}) {
        color: ${props => props.theme.colors.main};
      }
    }

    & i {
      animation: ${fadeInUp} ease-out 0.3s;

      @media only screen and (max-width: ${breakpoints.bpMedium}) {
        & > svg {
          fill: ${props => props.theme.colors.main};
        }
      }
    }

    position: relative;
  }

  .active::after {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    border-right: 0.5rem solid ${props => props.theme.colors.main};

    animation: ${fadeIn} 0.5s;

    @media only screen and (max-width: ${breakpoints.bpMedium}) {
      display: none;
    }
  }

  & p {
    display: none;
  }
`;

const Item = styled(NavLink)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 9.5rem;

  @media only screen and (max-width: ${breakpoints.bpMedium}) {
    height: 5rem;
  }
`;

const HomeIcon = styled(HomeOutlined)`
  font-size: 2.6rem;
  margin-bottom: 0.7rem;

  & svg {
    fill: ${props => props.theme.fontColors.textRgba(0.8)};
  }
`;

const CalendarIcon = styled(CalendarOutlined)`
  font-size: 2.6rem;
  margin-bottom: 0.7rem;

  & svg {
    fill: ${props => props.theme.fontColors.textRgba(0.8)};
  }
`;

const Text = styled.p`
  text-transform: uppercase;
  text-align: center;
  font-size: 1.2rem;

  font-weight: 500;
  color: ${props => props.theme.fontColors.textRgba(0.8)};

  margin: 0;
`;

export default Sidebar;
