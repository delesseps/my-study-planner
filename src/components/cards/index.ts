export { default as EvaluationCard } from "./EvaluationCard";
export { default as HomeworkCard } from "./HomeworkCard";
export { default as RecommendedActionCard } from "./RecommendedActionCard";
export { default as ToDoCard } from "./ToDoCard";
