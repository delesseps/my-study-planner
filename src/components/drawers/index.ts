export { default as EvaluationDrawer } from "./EvaluationDrawer";
export { default as HomeworkDrawer } from "./HomeworkDrawer";
export { default as ToDoDrawer } from "./ToDoDrawer";
export { default as CourseDrawer } from "./CourseDrawer";
