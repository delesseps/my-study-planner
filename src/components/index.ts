export { default as BannerAuth } from "./BannerAuth";
export { default as FadeIn } from "./FadeIn";
export { default as Loading } from "./Loading";
export { default as OAuthButtons } from "./OAuthButtons";
export { default as Section404 } from "./Section404";
export { default as Sidebar } from "./Sidebar";
export { default as SignInForm } from "./SignInForm";
export { default as SignUpForm } from "./SignUpForm";
export { default as TopBar } from "./TopBar";
export { default as UploadPicture } from "./UploadPicture";
export { default as FullPageErrorFallback } from "./FullPageErrorFallback";
export { default as Profiler } from "./Profiler";
