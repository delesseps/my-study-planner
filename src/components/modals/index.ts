export { default as EvaluationDescription } from "./EvaluationDescription";
export { default as HomeworkDescription } from "./HomeworkDescription";
export { default as Welcome } from "./Welcome";
export { default as UserProfile } from "./UserProfile";
