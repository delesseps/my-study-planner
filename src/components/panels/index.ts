export { default as CalendarPanel } from "./CalendarPanel";
export { default as CounterPanel } from "./CounterPanel";
export { default as EvaluationPanel } from "./EvaluationPanel";
export { default as HomeSchedulePanel } from "./HomeSchedulePanel";
export { default as HomeworkPanel } from "./HomeworkPanel";
export { default as RecommendedActionsPanel } from "./RecommendedActionsPanel";
export { default as ToDoPanel } from "./ToDoPanel";

