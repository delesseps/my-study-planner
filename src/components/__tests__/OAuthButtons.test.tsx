import React from "react";
import { render } from "test-utils";
import { OAuthButtons } from "components";

test("renders correctly", () => {
  const { asFragment } = render(<OAuthButtons type="signin" />);
  expect(asFragment()).toMatchSnapshot();
});

test("displays Sign in with Google", () => {
  const { getByText } = render(<OAuthButtons type="signin" />);
  getByText("Sign in with Google");
});

test("displays Sign up with Google", () => {
  const { getByText } = render(<OAuthButtons />);
  getByText("Sign up with Google");
});
