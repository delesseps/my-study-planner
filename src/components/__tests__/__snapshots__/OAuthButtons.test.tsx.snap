// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  margin-top: 2.4rem;
}

.c0 > a:not(:last-child) {
  margin-bottom: 2rem;
}

.c1.c1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 24px;
  margin-right: 1.2rem;
}

<div
    class="c0"
  >
    <a
      class="c1 ant-btn ant-btn-default ant-btn-lg ant-btn-block"
      href="http://localhost:3001/api/auth/google"
    >
      <svg
        class="c2"
      >
        google_logo.svg
      </svg>
      Sign in with Google
    </a>
  </div>
</DocumentFragment>
`;
